File: C,C++/baseball/baseball_11/Result.cpp
#include <iostream>
#include "Result.h"
#include "Constant.h"
#include "Answer.h"
#include "Guess.h"

void Result::calculate(Answer* answer, Guess* guess)
{
    for (int i = 0; i < DIGIT; i++)
    {
        int j = (i + 1) % DIGIT;
        int k = (i + 2) % DIGIT;

        if (guess->at(i) == answer->at(i))
            strike++;
        else if (guess->at(i) == answer->at(j) || guess->at(i) == answer->at(k))
            ball++;
        else
            out++;
    }
}

void Result::print()
{
    std::cout << "S:" << strike << " B:" << ball << " O:" << out << std::endl;
}

bool Result::isCorrect()
{
    return strike == DIGIT;
}

File: IoT기반_스마트팩토리_SW개발_전문가_과정/C#/baseball/Result.cs
using System;

namespace BaseballCSharp
{
    public class Result
    {
        private int _strike;
        private int _ball;
        private int _out;

        public void Calculate(Answer answer, Guess guess)
        {
            for (int i = 0; i < Constant.Digit; i++)
            {
                int j = (i + 1) % Constant.Digit;
                int k = (i + 2) % Constant.Digit;

                if (guess[i] == answer[i])
                    _strike++;
                else if (guess[i] == answer[j] || guess[i] == answer[k])
                    _ball++;
                else
                    _out++;
            }
        }

        public void Print()
        {
            Console.WriteLine($"S:{_strike} B:{_ball} O:{_out}");
        }

        public bool isCorrect()
        {
            return _strike == Constant.Digit;
        }
    }
}